reana/reana_dev/git.py

```diff
@@ -778,21 +778,44 @@ def git_upgrade(component, base):  # noqa: D301
             run_command(cmd, component)
 
 
+@click.argument("range", default="")
 @click.option(
     "--component",
     "-c",
     multiple=True,
     default=["CLUSTER"],
     help="Which components? [shortname|name|.|CLUSTER|ALL]",
 )
-@click.option("--number", "-n", default=6, help="Number of commits to output [6]")
+@click.option(
+    "--exclude-components",
+    default="",
+    help="Which components to exclude from build? [c1,c2,c3]",
+)
+@click.option("--number", "-n", default=10, help="Number of commits to output [10]")
+@click.option("--graph", is_flag=True, default=False, help="Show log graph?")
+@click.option("--oneline", is_flag=True, default=False, help="Show one-line format?")
+@click.option("--stat", is_flag=True, default=False, help="Show diff stat?")
+@click.option("--patch", "-p", is_flag=True, default=False, help="Show diff patch?")
 @click.option("--all", is_flag=True, default=False, help="Show all references?")
+@click.option("--no-pager", is_flag=True, default=False, help="Avoid output pager?")
```

Comment:
Minor: While keeping the same defaults as Git command makes sense, in this case, I would vote for having no pager by default. Pager output lacks context (I have default Git, no config).

To illustrate I've created a `test` commit on top of REANA-Client's `maint-0.7`:
- With pager executing `reana-dev git-log -c CLIENT master..maint-0.7`:
```console
commit d014498fb10e12c2306fac4fce3286ffa56d6413 (HEAD -> maint-0.7)
Author: Diego Rodriguez <diego.rodriguez@cern.ch>
Date:   Fri Nov 6 10:49:19 2020 +0100

    test
```
- Specifying `--no-pager`:
```console
$ reana-dev git-log -c CLIENT master..maint-0.7 --no-pager
[2020-11-06T10:58:06] reana-client: git --no-pager log -n 10 master..maint-0....
commit d014498fb10e12c2306fac4fce3286ffa56d6413 (HEAD -> maint-0.7)
Author: Diego Rodriguez <diego.rodriguez@cern.ch>
Date:   Fri Nov 6 10:49:19 2020 +0100

    test
[2020-11-06T10:58:06] reana-commons: git --no-pager log -n 10 master..maint-0....
[2020-11-06T10:58:07] reana-db: git --no-pager log -n 10 master..maint-0....
[2020-11-06T10:58:07] pytest-reana: git --no-pager log -n 10 master..maint-0....
```

Reply:
I agree, the pager is useful only for a handful of commands, e.g. `log -n 10 -p -c CLUSTER`, but these are not common... So we could have no pager by default and introduce new `--paginate` for pagination. (To keep harmony with `git log` command options.)